src/clr-angular/progress/progress-bars/progress-bar.ts

```diff
@@ -3,17 +3,27 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, Input, HostBinding } from '@angular/core';
+import { Component, Inject, Input, HostBinding, AfterViewChecked } from '@angular/core';
 import { isBooleanAttributeSet } from '../../utils/component/is-boolean-attribute-set';
+import { AriaLiveService, AriaLivePoliteness } from '../../utils/a11y/aria-live.service';
 
 @Component({
+  providers: [AriaLiveService],
   selector: 'clr-progress-bar',
   template: `
     <progress [id]="id" [attr.max]="max" [attr.value]="value" [attr.data-displayval]="displayValue"></progress>
-    <span *ngIf="displayAriaLive()" [attr.aria-live]="ariaLive">{{ displayValue }}</span>
+    <span *ngIf="displayAriaLive()">{{ displayValue }}</span>
   `,
 })
-export class ClrProgressBar {
+export class ClrProgressBar implements AfterViewChecked {
+  constructor(private ariaLiveService: AriaLiveService) {}
+
+  ngAfterViewChecked() {
+    if (this.displayAriaLive()) {
+      this.ariaLiveService.announce(this.displayValue, this.ariaLive);
+    }
+  }
+
```

Comment:
So I misunderstood how `ngAfterViewChecked` worked. I mistakenly thought it was only called on the component and its direct children but it can be called during any change detection cycle. 

![Screen Shot 2019-11-14 at 11 51 37 AM](https://user-images.githubusercontent.com/2021067/68884993-f1a4ca00-06d9-11ea-8612-5d74aa346f8a.png)

As you can see here as I click an unrelated button CD is called and the component is writing to the DOM several times. The fix I found to work locally is to remove the `ngAfterViewChecked` and instead turn the `value` property into a getter/setter combo so we can call the announcement only when the value changes. This results in write more consistent with this:

![Screen Shot 2019-11-14 at 12 19 06 PM](https://user-images.githubusercontent.com/2021067/68885126-2add3a00-06da-11ea-86e1-6e38e1f1268e.png)

```typescript
  private _value = 0;
  @Input('clrValue')
  get value() {
    return this._value;
  }

  set value(value) {
    this._value = value;
    if (this.displayAriaLive()) {
      this.ariaLiveService.announce(this.displayValue, this.ariaLive);
    }
  }
```

Reply:
Oh damn! Thanks for catching it @coryrylan. I will go and fix them right now!